### PR TITLE
[Refactor] rename Runs -> MultiRunManager and cleanup hooks

### PR DESCRIPTION
**Pull request by Jackmin801**
> [!NOTE]
> Modernizes multi-run management and propagates API changes across the codebase.
> 
> - Introduces global singleton `MultiRunManager` replacing `Runs` with explicit `discover_runs()` (master-only) and `synchronize_state()` (all ranks) and new master-only `discovered`/`forgotten` hooks; retains `creation`/`deletion` hooks and config validation
> - Centralizes LoRA rank validation and scaling in `setup_multi_run_manager(..., lora_config)`; modules now `register_with_runs(get_multi_run_manager(), ...)`
> - Updates trainer loop, optimizer/scheduler, checkpointing, broadcasters (filesystem/NCCL), data loader/packer, and transports to use `get_multi_run_manager()` and new ready/progress APIs
> - Simplifies per-run param/state access via `get_named_parameters_for_run()`, `get_state_dict_for_run()`, `reset_run_parameters()`
> - Renames and rewrites docs (`runs.md` → MultiRunManager) and adjusts `docs/index.md`; adds unit tests for run discovery/deletion/config handling
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit b951de9e43920d569fc6b33bf096cf9e26c795e2. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Review by samsja — APPROVED** [review]

File: src/prime_rl/trainer/rl/train.py [hunk 1]
```diff
@@ -126,11 +126,11 @@ def validate_lora_rank(orch_config) -> tuple[bool, str]:
...
 
-        def compute_scaling(orch_config) -> float:
-            return orch_config.model.lora.alpha / orch_config.model.lora.rank
+        def on_run_discovered(idx: int, run_id: str, orch_config) -> None:
+            multi_run_manager.scaling_factors[idx] = orch_config.model.lora.alpha / orch_config.model.lora.rank
 
-        runs.register_config_validation_hook(validate_lora_rank)
-        runs.register_scaling_hook(compute_scaling)
+        multi_run_manager.register_config_validation_hook(validate_lora_rank)
+        multi_run_manager.register_discovered_hook(on_run_discovered)
```
[the remark this answers is not in the document]
Comment (Jackmin801) [reply src/prime_rl/trainer/rl/train.py]: yea good catch. damn, we should rlly have multi node trainer tests soon
